packages/ui/src/proposals/components/ProposalDetails/ProposalDetails.tsx

```diff
@@ -72,6 +75,24 @@ export const ProposalDetails = ({ proposalDetails }: Props) => {
   const detailsRenderStructure = useMemo(() => getDetailsRenderStructure(proposalDetails), [proposalDetails])
 
   const additionalDetails = useMemo(() => {
+    if (gracePeriod) {
```

Comment:
Thanks for the heads up. I tried it and i think adding type `setGracePeriod` doesn't work because `gracePeriod` originally isn't a type of proposal just like `fundingRequest, cancelWorkingGroupLeadOpening...` and the other types.

In `pioneer/packages/ui/src/proposals/types/constants.ts` line 6. its already a type property of `ProposalConstants` so it is different from the `proposalDetails?.type`